cyclonedx/model/composition.py

```diff
@@ -0,0 +1,231 @@
+    """
+    INCOMPLETE_FIRST_PARTY_OPENSOURCE_ONLY = 'incomplete_first_party_opensource_only'
+
+    """
+    The relationship is incomplete. Only relationships for third-party components, services, or their dependencies are
+    represented.
+    """
+    INCOMPLETE_THIRD_PARTY_ONLY = 'incomplete_third_party_only'
+
+    """
+    The relationship is incomplete. Only relationships for third-party components, services, or their dependencies are
+    represented, limited specifically to those that are proprietary.
+    """
+    INCOMPLETE_THIRD_PARTY_PROPRIETARY_ONLY = 'incomplete_third_party_proprietary_only'
+
+    """
+    The relationship is incomplete. Only relationships for third-party components, services, or their dependencies are
+    represented, limited specifically to those that are opensource.
+    """
+    INCOMPLETE_THIRD_PARTY_OPENSOURCE_ONLY = 'incomplete_third_party_opensource_only'
+
+    """
+    The relationship may be complete or incomplete. This usually signifies a 'best-effort' to obtain constituent
+    components, services, or dependencies but the completeness is inconclusive.
+    """
+    UNKNOWN = 'unknown'
+
+    """
+    The relationship completeness is not specified.
+    """
+    NOT_SPECIFIED = 'not_specified'
+
+
+@serializable.serializable_class
+class CompositionReference:
+    """
+    Models a reference for an assembly or dependency in a Composition.
+
+    .. note::
+        See https://cyclonedx.org/docs/1.4/xml/#type_compositionType
+    """
+
+    def __init__(self, *, ref: BomRef) -> None:
+        self.ref = ref
+
+    @property
+    @serializable.json_name('.')
+    @serializable.type_mapping(BomRefHelper)
+    @serializable.xml_attribute()
+    def ref(self) -> BomRef:
+        """
+        References a component or service by its bom-ref attribute.
+
+        Returns:
+            `BomRef`
+        """
+        return self._ref
+
+    @ref.setter
+    def ref(self, ref: BomRef) -> None:
+        self._ref = ref
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, CompositionReference):
+            return hash(other) == hash(self)
+        return False
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, CompositionReference):
+            return self.ref < other.ref
+        return NotImplemented
+
+    def __hash__(self) -> int:
+        return hash(self.ref)
+
+    def __repr__(self) -> str:
+        return f'<CompositionReference ref={self.ref!r}>'
+
+
+@serializable.serializable_class
+class Composition:
+    """
+    This is our internal representation of the `compositionType` type within the CycloneDX standard.
+
+    .. note::
+        Introduced in CycloneDX v1.4
+
+    .. note::
+        See the CycloneDX Schema for hashType: https://cyclonedx.org/docs/1.4/xml/#type_compositionType
+    """
+
+    def __init__(self, *, aggregate: AggregateType, assemblies: Optional[Iterable[CompositionReference]] = None,
```

Comment:
`aggregate` has a default value: `not_specified`
this default value is missing here.

- see https://github.com/CycloneDX/specification/blob/8e131b1688ccfe41e1bfdd4b3280f33dcc06d04c/schema/bom-1.3.xsd#L1427
- see https://github.com/CycloneDX/specification/blob/8e131b1688ccfe41e1bfdd4b3280f33dcc06d04c/schema/bom-1.3.schema.json#L1031